### PR TITLE
Chore: Setup unit testing for backend

### PR DESCRIPTION
**Pull request by saszaCodes**
#### What does this PR do?
Sets up unit tests, adds one sample test and moves logic responsible for listening to new requests to separate file (this was done to prevent having unresolved promises when testing)

#### Description of Task to be completed?
Setup unit testing for backend

#### How should this be manually tested?
Run `npm run test` script. All checks should pass. After that, change `res.send('This is my first Express server');` to `res.sendStatus(500);` in `app.js` and run `npm run test` one more time. Now the test should fail.

#### Any background context you want to provide?
@Avuidrauxs you didn't specified which test suites to use, so I decided to go with supertest and jest.

#### Screenshots (if appropriate)
test passed

#### Questions:
Will we be working in TDD methodology?

### CODE REVIEW

**Comment by saszaCodes** [comment]
@Avuidrauxs as far as I can tell all of Hound's comment's are due to lack of `hound.yml` file, which I added in a branch that is at the moment waiting for your approval. ESLint doesn't throw any errors, only one warning about the console.log(), but I'm not removing it, because you seggested it be there

**Review by Avuidrauxs — APPROVED** [review]

**Pull request merged**
